frontend/web/components/pages/GitHubSetupPage.tsx

```diff
@@ -0,0 +1,286 @@
+    new URLSearchParams(location.search).get('installation_id') || ''
+  const [organisation, setOrganisation] = useState<string>('')
+  const [project, setProject] = useState<any>({})
+  const [projects, setProjects] = useState<projectType[]>([])
+  const [repositoryName, setRepositoryName] = useState<string>('')
+  const [repositoryOwner, setRepositoryOwner] = useState<string>('')
+  const [repositories, setRepositories] = useState<any>([])
+
+  const { data: repos, isSuccess: reposLoaded } = useGetGithubReposQuery(
+    {
+      installation_id: installationId,
+    },
+    { skip: !installationId },
+  )
+
+  const [
+    createGithubIntegration,
+    { data, isSuccess: isSuccessCreateGithubIntegration },
+  ] = useCreateGithubIntegrationMutation()
+  const baseUrl = window.location.origin
+
+  const [
+    createGithubRepository,
+    { isSuccess: isSuccessCreatedGithubRepository },
+  ] = useCreateGithubRepositoryMutation()
+
+  useEffect(() => {
+    if (reposLoaded && repos.repositories) {
+      setRepositoryOwner(repos?.repositories[0].owner.login)
+      setRepositories(repos)
+    }
+  }, [repos, reposLoaded])
+
+  useEffect(() => {
+    if (
+      isSuccessCreatedGithubRepository &&
+      !localStorage?.githubIntegrationSetupFromFlagsmith
+    ) {
+      window.location.href = `${baseUrl}/`
+    }
+  }, [isSuccessCreatedGithubRepository])
+
+  useEffect(() => {
+    if (
+      localStorage?.githubIntegrationSetupFromFlagsmith &&
+      localStorage?.githubIntegrationSetupFromFlagsmith === 'install'
+    ) {
+      createGithubIntegration({
+        body: {
+          installation_id: installationId,
+        },
+        organisation_id: JSON.parse(localStorage.lastEnv).orgId,
+      })
+    }
+  }, [])
+
+  useEffect(() => {
+    const createRepositories = async () => {
+      try {
+        if (
+          isSuccessCreateGithubIntegration &&
+          localStorage?.githubIntegrationSetupFromFlagsmith
+        ) {
+          const dataToSend = { 'installationId': installationId }
+          window.opener.postMessage(dataToSend, '*')
+        }
+        if (
+          data &&
+          isSuccessCreateGithubIntegration &&
+          !localStorage?.githubIntegrationSetupFromFlagsmith
+        ) {
+          const promises = []
+          for (const project of projects) {
+            const promise = createGithubRepository({
+              body: {
+                project: project.value,
+                repository_name: repositoryName,
+                repository_owner: repositoryOwner,
+              },
+              github_id: data.id,
+              organisation_id: organisation,
+            })
+            promises.push(promise)
+          }
+          await Promise.all(promises)
+          toast('Integration Created')
+        }
+      } catch (error) {
+        console.error('Error creating:', error)
+      }
+    }
+
+    createRepositories()
+  }, [
+    data,
+    isSuccessCreateGithubIntegration,
```

Comment:
This pattern leads to confusing, hard to read code. Instead, can we do 

createGithubIntegration({}).then((res)=>{
  if(!res.error){
   do stuff
  }
})

Reply:
Updated